### PR TITLE
Add FXIOS-22841 - Added waitForElementsToExist function

### PR DESCRIPTION
**Pull request by hemche**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10435)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22841)

## :bulb: Description

Currently, we have `mozWaitForElementToExist` which is being used in most of the tests ✅ 
For tests, where we are using `mozWaitForElementToExist` to verify multiple elements to be present on the screen, we are repeating it in sequence, like below
```
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.bookmarkTrayFill])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.download])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.readingList])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.login])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.sync])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.nightMode])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.whatsNew])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.helpCircle])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.edit])
        mozWaitForElementToExist(app.tables.otherElements[StandardImageIdentifiers.Large.settings])
```

It would be beneficial to have a `waitForElementsToExist` function to exist which will accept multiple elements and verify their presence on the screen at the same time in parallel and not in sequence. This will save execution time in larger test suites.

**Existing Behaviour:** It took 1 min 35 secs to verify all the elements present in the script
Screenshot 2024-10-31 at 11 50 20 AM

**After this change** It took 2 secs to verify all the elements present in the script
Screenshot 2024-10-31 at 11 26 47 AM

I have run full functional suite, smoke tests and verified the changes. There are no failures due to this change.

Screenshot 2024-10-31 at 12 28 47 AM
Screenshot 2024-10-31 at 12 16 43 AM
Screenshot 2024-10-30 at 11 59 01 PM
Screenshot 2024-10-30 at 11 50 29 PM

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by isabelrios** [comment]
@hemche thanks for your PR! so that we can run it please fix this swiftlint error: https://github.com/mozilla-mobile/firefox-ios/pull/22842/checks?check_run_id=32347787985

If I may, I would suggest creating other branch rather than main for PRs, thanks!

**Comment by hemche** [comment]
> @hemche thanks for your PR! so that we can run it please fix this swiftlint error: https://github.com/mozilla-mobile/firefox-ios/pull/22842/checks?check_run_id=32347787985
> 
> If I may, I would suggest creating other branch rather than main for PRs, thanks!

@isabelrios Fixed in https://github.com/mozilla-mobile/firefox-ios/pull/22842/commits/8fdc96ffffa42fea4b6680ba3a10006bbd9fb390
Not sure why my local swiftlint had 0 violations while pushing previously.
Sure, Will create feature branches instead of `main` going forward.

**Comment by isabelrios** [comment]
@hemche looks like there is one lint error still

**Comment by hemche** [comment]
Screenshot 2024-11-04 at 11 26 44 AM
@isabelrios I am not getting any error if i run `swiftlint` Let me know if there is a setup guide for swiftlint somewhere which i can follow to update my rules. Thank you!

**Comment by isabelrios** [comment]
> Screenshot 2024-11-04 at 11 26 44 AM @isabelrios I am not getting any error if i run swiftlint. Let me know if there is a setup guide for swiftlint somewhere which i can follow to update my rules. Thank you!

mm no, there is not any guide.. it should appear there.. which swiftlint version do you have?

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BaseTestCase.swift [hunk 1]
```diff
@@ -428,6 +427,24 @@ class BaseTestCase: XCTestCase {
         navigator.nowAt(SettingsScreen)
         app.buttons["Done"].waitAndTap()
     }
+    func waitForElementsToExist(_ elements: [XCUIElement], timeout: TimeInterval = TIMEOUT, message: String? = nil) {
+        var elementsDict = [XCUIElement: String]()
+        for element in elements {
+            elementsDict[element] = "exists == true"
+        }
+        let expectations = elementsDict.map(
+            {
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/BaseTestCase.swift:436]: The lint error is still in this line

Comment (hemche) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/BaseTestCase.swift]: @isabelrios Hoping this should fix the lint error. I have `0.57.0` installed through `homebrew` locally
Screenshot 2024-11-04 at 1 09 57 PM

**Review by isabelrios — APPROVED** [review]

**Pull request merged**